llumnix/global_scheduler/dispatch_scheduler.py

```diff
@@ -14,38 +14,63 @@
 from typing import Dict
 
 from llumnix.logging.logger import init_logger
-from llumnix.instance_info import InstanceInfo
+from llumnix.instance_info import InstanceType, InstanceInfo
 from llumnix.constants import DISPATCH_LOG_FREQUENCY
-from llumnix.global_scheduler.dispatch_policy import DispatchPolicyFactory
+from llumnix.global_scheduler.dispatch_policy import DispatchPolicyFactory, DispatchPolicy
 
 logger = init_logger(__name__)
 
 
 class DispatchScheduler:
     def __init__(self,
                  dispatch_policy: str,
-                 topk_random_dispatch: int) -> None:
-        self.dispatch_policy = DispatchPolicyFactory.get_policy(dispatch_policy)
-        self.topk_random_dispatch = topk_random_dispatch
+                 topk_random_dispatch) -> None:
+        self.dispatch_policy: DispatchPolicy = DispatchPolicyFactory.get_policy(
+            dispatch_policy, topk_random_dispatch=topk_random_dispatch)
 
         # statistics
         self.instance_type_num_requests: Dict[str, int] = {}
 
     def dispatch(
         self,
+        instance_type: InstanceType,
         instance_info: Dict[str, InstanceInfo],
-        instance_num_requests: Dict[str, int]
+        instance_num_requests: Dict[str, int],
     ) -> str:
-        instance_infos = list(instance_info.values())
-        instance_type = instance_infos[0].instance_type
         num_requests = self.instance_type_num_requests.get(instance_type, 0) + 1
         self.instance_type_num_requests[instance_type] = num_requests
         dispatch_instance_id = self.dispatch_policy.dispatch(
-            instance_num_requests, instance_infos, self.topk_random_dispatch
+            instance_num_requests, instance_info.values()
```

Comment:
why not just pass a complete instance_info? which is consistent with the instance_num_requests